sympy/printing/tests/test_ccode.py

```diff
@@ -571,7 +571,7 @@ def test_ccode_Assignment():
 
 def test_ccode_For():
     f = For(x, Range(0, 10, 2), [aug_assign(y, '*', x)])
-    assert ccode(f) == ("for (x = 0; x < 10; x += 2) {\n"
+    assert ccode(f) == ("for (int x = 0; x < 10; x += 2) {\n"
```

Comment:
This is a breaking change (in general we shouldn't change existing tests, they are there not only to ensure correctness, but also to guarantee a stable API for end users). Besides, I don't think that generating code for `For` should not imply declaration of the loop variable in C (it's not even valid C89, which is still what e.g. msvc targets and what most embedded systems use). You'll have to generate the declaration and initialization outside the printer...

Reply:
this is still a breaking change...